Rules/CommonScripts/KickAFK.as

```diff
@@ -68,17 +72,21 @@ void onTick(CRules@ this)
 	if (warned && time - warnTime > warnToKickSeconds)
 	{
 		//so long, sucker
-		client_AddToChat("You were " + (kickImmunity ? "moved to spectator" : "kicked") + " for being AFK too long.", SColor(255, 240, 50, 0));
+		client_AddToChat("You were " + (kickToSpectator ? "moved to spectator" : "kicked") + " for being AFK too long.", SColor(255, 240, 50, 0));
 		warned = false;
-		if (!kickImmunity)
+		if (!kickToSpectator)
 		{
 			getNet().DisconnectClient();
 		}
-		else
+		else if (kickImmunity)
 		{
-			joinNewSpecTeam(this, getLocalPlayer()); //Force-swap to spec team.
+			joinNewSpecTeam(this, p); //Force-swap to spec team.
```

Comment:
Does this need to be kept? The current way where admins need to type `!m` to rejoin their old team is annoying and the chat message just seems like a hacky workaround.

Reply:
All it does is put admins into moderator mode when afk kicked. Otherwise, it would just send admins into spectator without moderator visuals. I'm not sure which is preferred

>The current way where admins need to type !m to rejoin their old team is annoying

Changes to adminlogic should be done in another PR. put up an issue if you think its important

Reply:
My opinion is that there's no point in putting admins into moderator mode when they go AFK. 

But looking through the history, it looks like that was added so that players could be swapped to spec even if the game mode doesn't have a spec team. (#460)

I guess that raises the question of whether that issue will re-appear with this PR. Maybe instead of a blacklist a la `ignoredSeedingGamemodes`, there should be a whitelist of "seed-able" game modes?

Reply:
The concern wasn't with admins specifically, but with what happens if you try to swap any player to spec when there is no spec team. From testing in sandbox, it seems like it just respawns the player. Is that desirable? Not sure what might happen in mods.

It's likely a nonissue. Just floating some points of potential discussion.